features/poller/monitors/core/src/main/java/org/opennms/netmgt/poller/monitors/DNSResolutionMonitor.java

```diff
@@ -143,10 +143,10 @@ public PollStatus poll(MonitoredService svc, Map<String, Object> parameters) {
                     // hostname with port number
                     final Integer pos = nameserver.lastIndexOf(":");
                     final String hostname = nameserver.substring(0, pos);
-                    final String port = nameserver.substring(pos + 1);
+                    final Integer port = Integer.valueOf(nameserver.substring(pos + 1));
```

Comment:
`SystemProperties.getInteger()` should be used instead of `Integer.valueOf()` so we get better logging.

Reply:
That's the problem with the prior changes that this reverts. port isn't the name of a system property, so which of the static SystemProperties.getInteger() methods would be used here?

`public static Integer getInteger(String propertyName)`
`public static Integer getInteger(String propertyName, int defaultValue)`
`public static Integer getInteger(String propertyName, Integer defaultValue)`

Reply:
Oh! I see now, yeah.  That makes sense then.